content/intro-to-storybook/ember/en/simple-component.md

```diff
@@ -0,0 +1,223 @@
+---
+title: 'Build a simple component'
+tocTitle: 'Simple component'
+description: 'Build a simple component in isolation'
+---
+
+We’ll build our UI following a [Component-Driven Development](https://blog.hichroma.com/component-driven-development-ce1109d56c8e) (CDD) methodology. It’s a process that builds UIs from the “bottom up” starting with components and ending with screens. CDD helps you scale the amount of complexity you’re faced with as you build out the UI.
+
+## Task
+
+![Task component in three states](/intro-to-storybook/task-states-learnstorybook.png)
+
+`Task` is the core component in our app. Each task displays slightly differently depending on exactly what state it’s in. We display a checked (or unchecked) checkbox, some information about the task, and a “pin” button, allowing us to move tasks up and down the list. Putting this together, we’ll need these props:
+
+- `title` – a string describing the task
+- `state` - which list is the task currently in and is it checked off?
+
+As we start to build `Task`, we first write our test states that correspond to the different types of tasks sketched above. Then we use Storybook to build the component in isolation using mocked data. We’ll “visual test” the component’s appearance given each state as we go.
+
+This process is similar to [Test-driven development](https://en.wikipedia.org/wiki/Test-driven_development) (TDD) that we can call “[Visual TDD](https://blog.hichroma.com/visual-test-driven-development-aec1c98bed87)”.
+
+## Get setup
+
+First, let’s create the necessary files for our task component and its accompanying story file: `app/components/Task.hbs` and `app/components/Task.stories.js`.
+
+We’ll begin with a basic implementation of the `Task`, simply taking in the attributes we know we’ll need and the two actions you can take on a task (to move it between lists):
+
+```handlebars
+
+{{!--  app/components/Task.hbs --}}
+
+<div class="list-item">
+  <input type="text" value={{@task.title}} readonly={{true}}/>
+</div>
+```
+
+Above, we render straightforward markup for `Task` based on the existing HTML structure of the Todos app.
+
+Below we build out Task’s three test states in the story file:
+
+```javascript
+//  app/components/Task.stories.js
+import { hbs } from 'ember-cli-htmlbars';
+import { action } from '@storybook/addon-actions';
+
+export default {
+  title: 'Task',
+  component: 'Task',
+  excludeStories: /.*Data$/,
+};
+
+export const taskData = {
+  id: '1',
+  title: 'Test Task',
+  state: 'TASK_INBOX',
+  updatedAt: new Date(2018, 0, 1, 9, 0),
+};
+
+export const actionsData = {
+  onPinTask: action('onPinTask'),
+  onArchiveTask: action('onArchiveTask'),
+};
+
+export const Default = () => ({
+  template: hbs`{{Task task=task pinTask=(action taskActions.onPinTask) archiveTask=(action taskActions.onArchiveTask)}}`,
```

Comment:
```suggestion
  template: hbs`<Task @task={{task}} @pinTask={{fn this.pinTask}} @archiveTask={{fn this.archiveTask}}/>`,
```

Reply:
@gossi what you're proposing will not work. I tried it and some variations to see if it would match the current state of both frameworks.

To give you a bit more context about this.

Having the code i'm currently using, more specifically:
```js
import { hbs } from 'ember-cli-htmlbars';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Task',
  component: 'Task',
  excludeStories: /.*Data$/,
};

export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updatedAt: new Date(2018, 0, 1, 9, 0),
};

export const actionsData = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask'),
};

export const Default = () => ({
  template: hbs`{{Task task=task pinTask=(action taskActions.onPinTask) archiveTask=(action taskActions.onArchiveTask)}}`,
  
  context: {
    task: {
      ...taskData,
    },
    taskActions: {
      ...actionsData,
    }, 
  },
});
``` 

It will yeld the following:

![gossi_feedback_simple_component_v1](https://user-images.githubusercontent.com/22988955/81019804-51955580-8e5f-11ea-8c5f-df4654196e0a.png)

Making the change you're proposing:
```js
import { hbs } from 'ember-cli-htmlbars';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Task',
  component: 'Task',
  excludeStories: /.*Data$/,
};

export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updatedAt: new Date(2018, 0, 1, 9, 0),
};

export const actionsData = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask'),
};

export const Default = () => ({
  template:hbs`<Task @task={{this.task}} @pinTask={{fn this.actionsData.onPinTask}} @archiveTask={{fn this.actionsData.onArchiveTask}}/>`,
  context: {
    task: {
      ...taskData,
    },
    ...actionsData
  },
});
```

yelds the following:

![gossi_feedback_simple_component_v2](https://user-images.githubusercontent.com/22988955/81019970-b8b30a00-8e5f-11ea-800e-126058aecb6e.png)

You might argue that, ok the issue is with naming of the component. Changing the code to:
```js
import { hbs } from 'ember-cli-htmlbars';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Task',
  component: 'Task',
  excludeStories: /.*Data$/,
};

export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updatedAt: new Date(2018, 0, 1, 9, 0),
};

export const actionsData = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask'),
};

export const Default = () => ({
  template:hbs`<task @task={{this.task}} @pinTask={{fn this.actionsData.onPinTask}} @archiveTask={{fn this.actionsData.onArchiveTask}}/>`,
  context: {
    task: {
      ...taskData,
    },
    ...actionsData
  },
});
```
will yeld the following:

![gossi_feedback_simple_component_v3](https://user-images.githubusercontent.com/22988955/81020429-b1403080-8e60-11ea-8b13-e0391754fe83.png)

Various permutations of said code will yeld the same result to the extent of just having this:
 ```js
export const Default = () => ({
  template:hbs`<task @task={{this.task}} @pinTask={{fn this.pinTask}} @archiveTask={{fn this.archiveTask}}/>`,
  context: {
    task: {
      ...taskData,
    },
    pinTask:(e)=>console.log(e),
    archiveTask:(e)=>console.log(e)
  },
});
```
Also take a look at [this](https://storybook.js.org/docs/guides/guide-ember/) and check how the component is being implemented. Which applies to this case.

I'm going to pick up on this tomorrow (tuesday) and see the remainder proposed changes